src/App.tsx

```diff
@@ -1,18 +1,41 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const initialItems = getNumbers(1, 42);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [total] = useState(42);
+
+  const changePage = useCallback((page: number) => (
+    setCurrentPage(page)
+  ), []);
```

Comment:
It's basically the same function, as setCurrentPage. So you don't need to create additional function for it. UseState works like useCallback, when you create a method to manipulate some variable.